src/Identity/IdentityServer/BaseRequestValidator.cs

```diff
@@ -465,11 +495,19 @@ private Device GetDeviceFromRequest(ValidatedRequest request)
                     CoreHelpers.CustomProviderName(type));
                 if (type == TwoFactorProviderType.Duo)
                 {
-                    return new Dictionary<string, object>
+                    var duoResponse = new Dictionary<string, object>
                     {
                         ["Host"] = provider.MetaData["Host"],
                         ["Signature"] = token
                     };
+
+                    // DUO SDK v4 Update: Duo-Redirect
+                    if (FeatureService.IsEnabled(FeatureFlagKeys.DuoRedirect, CurrentContext))
+                    {
+                        // Use generate token from new token provider
```

Comment:
```suggestion
                        // Generate auth url from new token provider
```
Hopefully adds some value to the comment, ignore if its not what you meant